transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java

```diff
@@ -68,23 +68,32 @@ public View getView(@NonNull final ViewGroup viewGroup) {
                 @Override
                 public void onFocusChange(View v, boolean hasFocus) {
                     if (!hasFocus) {
-                        mValue = ((EditText) v).getText().toString();
+                        String input = ((EditText) v).getText().toString();
+                        mValue = formatToApi(input);
                         mListener.valueChanged();
                     } else {
                         mListener.widgetFocused(PhoneWidget.this.getName());
                     }
                 }
             });
             editText.addTextChangedListener(new TextWatcher() {
```

Comment:
Is this TextWatcher the same to all subclasses of AbstractMaskedInputWidget? If so, please move it there

Reply:
TextWidget, Numberwidget, PhoneWidget have their own private variables
```    private ViewGroup mContainer;
    private TextInputLayout mTextInputLayout;
    private String mValue;```

I'd have to move these to AbstractMaskedInputWidget as well. Do you want to do that?

Reply:
Please do it. You can move them to AbstractMaskedInputWidget and expose setters as package default (no modifier)